File: solutions/python/rna-transcription/1/rna_transcription.py
def to_rna(dna_strand: str) -> str:
    """Compute the RNA complement of a given DNA sequence

    Arguments:
        dna_strand -- DNA sequence

    Returns:
        RNA sequence
    """
    dna_to_rna_dict = {
        'G': 'C',
        'C': 'G',
        'T': 'A',
        'A': 'U'
    }

    return ''.join(map(lambda char: dna_to_rna_dict[char], dna_strand))

File: solutions/ruby/rna-transcription/1/rna_transcription.rb
module Complement
  DNA_TO_RNA = {
    'G' => 'C',
    'C' => 'G',
    'T' => 'A',
    'A' => 'U'
  }.freeze
  def self.of_dna(dna)
    dna.chars.map{|c| DNA_TO_RNA[c] || ''}.join
  end
end